napari/components/add_layers_mixin.py

```diff
@@ -155,91 +159,65 @@ def add_image(
         elif data is None:
             data = io.magic_imread(path)
 
+        # doing this here for IDE/console autocompletion in add_image function.
+        kwargs = {
+            'rgb': rgb,
+            'is_pyramid': is_pyramid,
+            'colormap': colormap,
+            'contrast_limits': contrast_limits,
+            'gamma': gamma,
+            'interpolation': interpolation,
+            'rendering': rendering,
+            'iso_threshold': iso_threshold,
+            'attenuation': attenuation,
+            'name': name,
+            'metadata': metadata,
+            'scale': scale,
+            'translate': translate,
+            'opacity': opacity,
+            'blending': blending,
+            'visible': visible,
+        }
+
         if channel_axis is None:
-            if colormap is None:
-                colormap = 'gray'
-            if blending is None:
-                blending = 'translucent'
-            layer = layers.Image(
-                data,
-                rgb=rgb,
-                is_pyramid=is_pyramid,
-                colormap=colormap,
-                contrast_limits=contrast_limits,
-                gamma=gamma,
-                interpolation=interpolation,
-                rendering=rendering,
-                iso_threshold=iso_threshold,
-                attenuation=attenuation,
-                name=name,
-                metadata=metadata,
-                scale=scale,
-                translate=translate,
-                opacity=opacity,
-                blending=blending,
-                visible=visible,
-            )
-            self.add_layer(layer)
-            return layer
+            kwargs['colormap'] = kwargs['colormap'] or 'gray'
+            kwargs['blending'] = kwargs['blending'] or 'translucent'
+            return self.add_layer(layers.Image(data, **kwargs))
         else:
-            if is_pyramid:
-                n_channels = data[0].shape[channel_axis]
-            else:
-                n_channels = data.shape[channel_axis]
-
-            name = ensure_iterable(name)
-
-            if blending is None:
-                blending = 'additive'
-
-            if colormap is None:
-                if n_channels < 3:
-                    colormap = colormaps.MAGENTA_GREEN
+            n_channels = (data[0] if is_pyramid else data).shape[channel_axis]
+            kwargs['blending'] = kwargs['blending'] or 'additive'
+
+            # turn the kwargs dict into a mapping of {key: iterator}
+            # so that we can use {k: next(v) for k, v in kwargs.items()} below
+            for key, val in kwargs.items():
+                if key == 'colormap' and val is None:
+                    if n_channels < 3:
+                        kwargs[key] = iter(colormaps.MAGENTA_GREEN)
+                    else:
+                        kwargs[key] = itertools.cycle(colormaps.CYMRGB)
+                elif key in {
```

Comment:
Can you add a comment on why these ones get special treatment (I guess because they themselves can be iterable)

Reply:
will do.  yeah, those are the 4 args that are already iterable even in the single channel case.